[2025-03-10T19:55:10.982+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: project2.insert_employees manual__2025-03-10T19:55:07.589200+00:00 [queued]>
[2025-03-10T19:55:10.998+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: project2.insert_employees manual__2025-03-10T19:55:07.589200+00:00 [queued]>
[2025-03-10T19:55:10.999+0000] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2025-03-10T19:55:10.999+0000] {taskinstance.py:1280} INFO - Starting attempt 1 of 1
[2025-03-10T19:55:11.000+0000] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2025-03-10T19:55:11.021+0000] {taskinstance.py:1300} INFO - Executing <Task(PostgresOperator): insert_employees> on 2025-03-10 19:55:07.589200+00:00
[2025-03-10T19:55:11.030+0000] {standard_task_runner.py:55} INFO - Started process 7280 to run task
[2025-03-10T19:55:11.035+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'project2', 'insert_employees', 'manual__2025-03-10T19:55:07.589200+00:00', '--job-id', '147', '--raw', '--subdir', 'DAGS_FOLDER/project2.py', '--cfg-path', '/tmp/tmp0xshy1z3']
[2025-03-10T19:55:11.036+0000] {standard_task_runner.py:83} INFO - Job 147: Subtask insert_employees
[2025-03-10T19:55:11.134+0000] {task_command.py:388} INFO - Running <TaskInstance: project2.insert_employees manual__2025-03-10T19:55:07.589200+00:00 [running]> on host f1d7771c7032
[2025-03-10T19:55:11.238+0000] {taskinstance.py:1509} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=project2
AIRFLOW_CTX_TASK_ID=insert_employees
AIRFLOW_CTX_EXECUTION_DATE=2025-03-10T19:55:07.589200+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2025-03-10T19:55:07.589200+00:00
[2025-03-10T19:55:11.239+0000] {sql.py:254} INFO - Executing: CALL update_employees();
[2025-03-10T19:55:11.253+0000] {base.py:73} INFO - Using connection ID 'postgres_conn' for task execution.
[2025-03-10T19:55:11.494+0000] {base.py:73} INFO - Using connection ID 'postgres_conn' for task execution.
[2025-03-10T19:55:11.526+0000] {sql.py:375} INFO - Running statement: CALL update_employees();, parameters: None
[2025-03-10T19:55:11.535+0000] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/operators/sql.py", line 266, in execute
    **extra_kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 349, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 380, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.NotNullViolation: null value in column "employee_id" of relation "employees" violates not-null constraint
DETAIL:  Failing row contains (null, 41, Female, Yes, Life Sciences, 3).
CONTEXT:  SQL statement "INSERT INTO employees (employee_id, age, gender, OverTime, EducationField, PerformanceRating)
SELECT s.employee_id, s.age, s.gender, s.OverTime, s.EducationField, s.PerformanceRating
FROM staging_hr s
LEFT JOIN employees e ON s.employee_id = e.employee_id WHERE e.employee_id IS NULL"
PL/pgSQL function update_employees() line 3 at SQL statement

[2025-03-10T19:55:11.547+0000] {taskinstance.py:1323} INFO - Marking task as FAILED. dag_id=project2, task_id=insert_employees, execution_date=20250310T195507, start_date=20250310T195510, end_date=20250310T195511
[2025-03-10T19:55:11.566+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 147 for task insert_employees (null value in column "employee_id" of relation "employees" violates not-null constraint
DETAIL:  Failing row contains (null, 41, Female, Yes, Life Sciences, 3).
CONTEXT:  SQL statement "INSERT INTO employees (employee_id, age, gender, OverTime, EducationField, PerformanceRating)
SELECT s.employee_id, s.age, s.gender, s.OverTime, s.EducationField, s.PerformanceRating
FROM staging_hr s
LEFT JOIN employees e ON s.employee_id = e.employee_id WHERE e.employee_id IS NULL"
PL/pgSQL function update_employees() line 3 at SQL statement
; 7280)
[2025-03-10T19:55:11.610+0000] {local_task_job.py:208} INFO - Task exited with return code 1
[2025-03-10T19:55:11.627+0000] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
